src/core/executors/ExecutorBase.ts

```diff
@@ -188,4 +189,20 @@ export default abstract class ExecutorBase {
       this._processOperationQueue.call(this);
     }
   }
+
+  private _prepareforExecution(operation: Operation<SupportedModel>): void {
```

Comment:
I like that when you added more code you split this out to keep functions smaller, however I think this function name is misleading. It is doing more than prepare, it also executes.

One suggestion that comes to mind is rename this function to `canExecute` returning a boolean. Then move  `this._processOperation` back to the while loop in `_processOperationQueue`.